src/tir/transforms/lower_tvm_builtin.cc

```diff
@@ -113,16 +113,6 @@ class BuiltinLower : public StmtExprMutator {
     op = stmt.as<AllocateNode>();
     // Get constant allocation bound.
     int64_t nbytes = GetVectorBytes(op->dtype);
-    if (device_type_.defined()) {
```

Comment:
cc @areusch @mbrookhart this is the code we discussed yesterday, currently CI passes with this code? that seems to contradict the conversations we had, but also I think these should be stack allocations for performance on cloud cpus.

Reply:
Yes, if we want the allocates for to stack in CPU PrimFuncs, maybe we should gate them to have storage_scope = 'local' and we should generate TVMBAWs for 'global' allocates.

Reply:
Filed https://github.com/apache/tvm/issues/9022.
Just checking your use case: You want to see the allocas so you can later rewrite them, right? It's a bit confusing because I though if the final c code generator sees these Allocates it rewrites them to use globals which I assumed was what you wanted.